### DISCUSSION TITLE
New Layouts: Unable to use own headers/footers and entire page space

### DISCUSSION THREAD

**Discussion by sergey-dme** [Budibase/budibase#1968] [Help] [answered]
Hi all, 

In new layouts, even with setting Navigation: none, It is unable to use the whole entire page space as before, moreover It is unable to fix by custom CSS such as:

display: flex;
max-width: 100%;
min-height: 100%;
flex-direction: column;
padding: 0 px;
margin: 0px;

It makes impossible for user to use user-custom headers/footers with settings width 100%. and whole page height: 100%
This issue exists only for newly created Layouts. Fortunately, my previously created ones works otherwise it would be catastrophic for me. 

And I don't have any free "old-empty" layouts anymore, all are already used. and want to create new empty.. with abilities to control content. 

The user-customzied header is not only use case. Another one: using layouts as Form-templates for printing. 

In new layouts, three options for "Width" not enough, there is always a gap/margin:

<img width="1532" alt="Screenshot 2021-07-10 at 13 07 46" src="https://user-images.githubusercontent.com/53024832/125159588-07a1b100-e181-11eb-9322-997c453aaeb8.png">

Please make possible to use and control the whole layout page space, as before without such restrictions. 

<img width="1527" alt="Screenshot 2021-07-10 at 13 07 27" src="https://user-images.githubusercontent.com/53024832/125159626-39b31300-e181-11eb-9c85-86d03368bc34.png">

**Comment by sergey-dme** [comment]
I think the solution is: option to choose when creating new layout: 
DEFAULT (current one layout in latest version) OR **BLANK** (empty as before just with the Screenslot).

**Comment by sergey-dme** [comment]
I kindly ask you to help me with this. Thank you.

**Comment by aptkingston** [comment, marked as answer]
@sergey-dme if you need this functionality you can simply apply a negative margin to your custom header. Here's a screenshot where I've added a container and used negative margin on it to achieve the same thing.

![image](https://user-images.githubusercontent.com/9075550/125160423-b2f83900-e174-11eb-88a3-44715009e1a0.png)

Having said that, perhaps it would be better to not have any padding on the empty layout, since the empty layout is probably only going to be used on pages with custom navigation bars. I'll discuss this with the team and we might change this.

**Reply by sergey-dme** [reply]
@aptkingston Thank you Andrew, It would be greatly appreciated. And thanks for temporary solution with negative margin, I new to CSS just started learn it recently when old design section was removed)) 
Yes, empty layouts are very important, and not only for custom headers. I also use them as Templates for printing Forms.

**Reply by aptkingston** [reply]
@sergey-dme after further thought, I think you're correct that empty layouts should not have padding - they should be a blank canvas. There should also be an option for the width to fill the screen. I've created a pull request for this functionality, so keep an eye on that towards the start of next week for updates 👍 
https://github.com/Budibase/budibase/pull/1969

**Reply by sergey-dme** [reply]
@aptkingston I hope the coming update will also fix a problem with overflow and page height for app created in prev. versions. Just imported my app, and no scrolling, footer is gone. It can be fix playing with overflow: auto and height 100vh, as you mentioned it in another ticket. But the result is not the same as in original app.
So  I am still staying on my original app, will wait for fixes and then re-import it.

**Reply by aptkingston** [reply]
@sergey-dme no we won't be making changes for scrolling, because layouts in the previous versions were just containers, and we cannot add scroll rules for all containers. There are 2 easy solutions which I have already described:

1. Update to the new layout component. Make a new layout, copy and paste your custom navigation, and update your screens to use this new layout.
2. Use CSS rules on your old layout. You said you've already fixed the scrolling by adding the CSS rules I mentioned, so I'm not sure what the problem is. What do you mean it is not the same as in the original app?

Both of those solutions are quick and easy, but if you need help doing it then let me know.

On another note, in the next update you will not need to export/import to update your app any more. You will be able to update apps to new versions (and downgrade back to the previous verison after updating if you need to) from inside the builder - so it will be much easier for you to try new updates.

**Reply by sergey-dme** [reply]
@aptkingston Okay, that sounds good, it is like App version control.  Great news!
Yep, to exclude any possible issues related with old containers used I will definitely re-create my layouts after the fix for empty ones without padding. 
Thank you.

**Reply by Manishkparmar80** [reply]
> @aptkingston Thank you Andrew, It would be greatly appreciated. And thanks for temporary solution with negative margin, I new to CSS just started learn it recently when old design section was removed)) 
> Yes, empty layouts are very important, and not only for custom headers. I also use them as Templates for printing Forms.

@sergey-dme, you have mentioned about , you r using Templates for printing forms. Can you pls share more detail of how you have configured budibase app to print forms.

**Comment by sergey-dme** [comment]
@aptkingston Could you please tell me when the fix with "Max" option will be released?.

**Reply by shogunpurple** [reply]
@sergey-dme this one was merged into our `develop` branch so it hasn't been released yet, but I can push and release that specific fix now 👍

**Reply by sergey-dme** [reply]
@aptkingston Thank you! Greatly appreciated!
